### PR TITLE
added README for dom equals API

### PR DESCRIPTION
**Pull request by desaikd**
*Description of changes:*
This PR adds description and usage for equals API of `dom.Value`.

*Changes:*
- description of `equals` and `ionEquals` methods for `dom.Value`.
- added examples for usage of both methods.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/dom/README.md [hunk 1]
```diff
@@ -264,3 +264,140 @@ assert.isTrue(string === "" + ionString);
 assert.isTrue(string === ionString.stringValue());
 assert.isFalse(ionString === ionString2);
 ```
+
+### `dom.Value` equivalence
+
+dom.Value object equivalence is implemented by two methods ionEquals(compares two dom.Value objects based on strict or non-strict comparison type) and equals(compares a dom.Value object with a corresponding JS object).
```
Comment (jobarr-amzn) [inline src/dom/README.md:270]: We ought to be using ``` `dom.Value` ```, ``` `ionEquals(compares two dom.Value objects based on strict or non-stric comparison type)` ```, I think. Code-formatted.

File: src/dom/README.md [hunk 2]
```diff
@@ -264,3 +264,140 @@ assert.isTrue(string === "" + ionString);
...
+
+#### `ionEquals`:
+
+This method compares a dom.Value object with another dom.Value object with either strict or non-strict comparison type. 
+
+*Strict equivalence* refers to Ion data model equivalence as defined in Ion Equivalence (https://www.javadoc.io/doc/com.amazon.ion/ion-java/latest/com/amazon/ion/util/Equivalence.html) and by Ion Specification (http://amzn.github.io/ion-docs/docs/spec.html)
+*Structural or non-strict equivalence* follows the same rules as strict equivalence,except that
```
Comment (jobarr-amzn) [inline src/dom/README.md:277]: Should these be links? E.g. [Ion Equivalence](https://www.javadoc.io/doc/com.amazon.ion/ion-java/latest/com/amazon/ion/util/Equivalence.html)?

File: src/dom/README.md [hunk 3]
```diff
@@ -264,3 +264,140 @@ assert.isTrue(string === "" + ionString);
...
+float1.ionEquals(float2); // returns true
+float1.ionEquals(float3); // return false as annotations didn't match in a strict equivalence
+
+float1.ionEquals(float4); // returns false as Decimal and Float values will not be equivalent
+float3.ionEquals(float5, {epsilon: 0.5}); // returns true as both values are equivalent by epsilon value
+float3.ionEquals(float5, {epsilon: 0.2}); // returns false as both values are not equivalent by epsilon value
+
+// Equivalence between JS Value and Ion DOM Value
+float1.equals(1.5); // returns true
+float1.ionEquals(1.5); // returns false
```
Comment (jobarr-amzn) [inline src/dom/README.md:320]: I think this particularly this example (and also the preceding one) would be clearer without variables. Reading this I'm bouncing back and forth constantly from definitions for `floatN` and their uses.

If we must have variables names, let the names be descriptive and helpful. `same1`, `same2`, `unannotated`, `ion_float`, `ion_decimal`, `nearby_float`... something like that?

File: src/dom/README.md [hunk 4]
```diff
@@ -264,3 +264,140 @@ assert.isTrue(string === "" + ionString);
...
+let list2: Value = load('planets::["Mercury", "Venus", "Earth", "Mars"]');
+let list3: Value = load('planets::["Mercury", "Venus"]');
+
+// Equivalence between two Ion DOM Values
+list1.ionEquals(list2) // returns true
+list1.ionEquals(list3) // returns false because the lists have different length
+
+// Equivalence between JS Value and Ion DOM Value
+list1.equals(["Mercury", "Venus", "Earth", "Mars"]); // returns true
+list1.equals(["Mercury", "Venus", "Earth"]); // returns false because the lists differ with one lement "Mars"
```
Comment (jobarr-amzn) [inline src/dom/README.md:352]: typo: element

File: src/dom/README.md [hunk 5]
```diff
@@ -264,3 +264,140 @@ assert.isTrue(string === "" + ionString);
...
+```
+
+####Struct Example:
+```javascript
+let struct1: Value = load(
+  "foo::bar::{" +
+    "name: {" +
+    'first: "John", ' +
+    'middle: "Jacob", ' +
+    'last: "Jingleheimer-Schmidt",' +
```
Comment (jobarr-amzn) [inline src/dom/README.md:362]: 😀

File: src/dom/README.md [hunk 6]
```diff
@@ -264,3 +264,140 @@ assert.isTrue(string === "" + ionString);
...
+struct1.equals(
+    {
+      name: {
+        first: "John",
+        middle: "Jacob",
+        last: "Jingleheimer-Schmidt",
+      },
+      age: 41,
+    }
+); // retrurns true
```
Comment (jobarr-amzn) [inline src/dom/README.md:402]: typo: returns

File: src/dom/README.md (hunk 3 above)
Comment (desaikd) [reply src/dom/README.md:320]: Yes, I think we can have variable names descriptive for making it more readable. I will add this change along with other typos and minor changes.

File: src/dom/README.md [hunk 7]
```diff
@@ -264,3 +264,140 @@ assert.isTrue(string === "" + ionString);
...
+
+*Strict equivalence* refers to Ion data model equivalence as defined in Ion Equivalence (https://www.javadoc.io/doc/com.amazon.ion/ion-java/latest/com/amazon/ion/util/Equivalence.html) and by Ion Specification (http://amzn.github.io/ion-docs/docs/spec.html)
+*Structural or non-strict equivalence* follows the same rules as strict equivalence,except that
+
+* 1. Annotations are not considered, and
+* 2. Timestamps that represent the same instant in time are always considered equivalent.
+
+#### `equals`:
+
+This method compares a dom.Value object with a corresponding JS object.
```
Comment (zslayton) [inline src/dom/README.md:284]: I think you should go into a bit more detail here.
* How are annotations handled? (I think they're ignored, right?) 
* For Ion types that don't have an obvious mapping to JS, what types can be considered equal? 
   * Is an `ion.dom.Blob` equal to an `Array`? A `Utf8Array`? A `Buffer`?)
   * Can I test an `ion.dom.Decimal` for equality with a `Number`?
   * etc

File: src/dom/README.md [hunk 8]
```diff
@@ -267,94 +267,111 @@ assert.isFalse(ionString === ionString2);
...
-#### Integer Example:
+This method compares a `dom.Value` object with a corresponding JS object as defined in [Ion Data Types section](#iondom-data-types). 
+If the other object is a `dom.Value` instance then this comparison checks for structural (or "non-strict") equivalence.
+
+_NOTE:_ 
+
+* This method also provides an optional parameter `epsilon` for `Float` type comparison.
+* This method allows coerce numeric types comparison for `Decimal` and `Float` values:
+    * Comparison of `Float` with `Decimal` or `number` type is allowed.
+    * Comparison of `Decimal` with `Float` or `number` type is allowed.
```
Comment (zslayton) [inline src/dom/README.md:301]: Can this be simplified to:
```suggestion
* This method allows `Decimal`, `Float`, and `number` values to be compared to one another.
```
?

Does it _not_ allow `Integer` values to be compared to the others?

File: src/dom/README.md [hunk 9]
```diff
@@ -267,94 +267,111 @@ assert.isFalse(ionString === ionString2);
...
-* 1. Annotations are not considered, and
-* 2. Timestamps that represent the same instant in time are always considered equivalent.
+Optional parameters that can be used with this method:
 
-#### `equals`:
+| option                      | description                                                                    |
+| --------------------------- | :------------------------------------------------------------------------------|
+|`epsilon`                    | used by `Float` for an equality with given epsilon precision. (Default: `null`)|
+|`ignoreAnnotations`          | specifies whether to ignore annotations or not for equality. (Default: `false`)|
+|`ignoreTimestampPrecision`   | specifies whether to ignore timestamp local offset and precision or not for equality. (Default: `false`)|
```
Comment (zslayton) [inline src/dom/README.md:289]: Consider making `default` its own column.

File: src/dom/README.md [hunk 10]
```diff
@@ -267,94 +267,111 @@ assert.isFalse(ionString === ionString2);
...
 
 *Strict equivalence* refers to Ion data model equivalence as defined in Ion Equivalence (https://www.javadoc.io/doc/com.amazon.ion/ion-java/latest/com/amazon/ion/util/Equivalence.html) and by Ion Specification (http://amzn.github.io/ion-docs/docs/spec.html)
-*Structural or non-strict equivalence* follows the same rules as strict equivalence,except that
+*Structural* (or *non-strict*) *equivalence* follows the same rules as strict equivalence,except that:
+* Annotations are not considered, and
+* Timestamps that represent the same instant in time are always considered equivalent.
 
-* 1. Annotations are not considered, and
-* 2. Timestamps that represent the same instant in time are always considered equivalent.
+Optional parameters that can be used with this method:
```
Comment (zslayton) [inline src/dom/README.md:283]:
```suggestion
This method accepts an optional configuration object that allows you to modify how equivalence is determined. It recognizes the following properties:
```

File: src/dom/README.md (hunk 8 above)
Comment (desaikd) [reply src/dom/README.md:301]: No it only allows `number` and `JSBI` with `Integer`.

Comment (zslayton) [reply src/dom/README.md]: That's surprising to me--I expected value-based `equals` to allow `Integer`-to-`Float`/`Decimal` comparisons. I've opened #696 to track this.

**Review by zslayton — APPROVED** [review]

**Pull request merged**